src/js/core/methods/CardanoSignTransaction.js

```diff
@@ -157,6 +171,10 @@ export default class CardanoSignTransaction extends AbstractMethod {
                 this._ensureFeatureIsSupported('MultiassetOutputs');
             }
         });
+
+        if (params.auxiliary_data) {
+            this._ensureFeatureIsSupported('AuxiliaryData');
```

Comment:
What happens if someone has new trezor connect, but old trezor firmware?

Reply:
Current Trezor fw enforces the old format of metadata (it parses the blob and ensures it's a dictionary) and throws if the new one (tuple) is passed, so we went with disabling metadata support for past firmwares completely. Mainstream tools use the new format exclusively anyway

Reply:
Regarding what exactly happens - if you pass the "metadata" field, connect throws as mentioned in https://github.com/vacuumlabs/connect/pull/12/files/3f0f5068956dfa76cace782504e4edf5d886f5ef#r609512216 and if you pass auxiliary data to an older fw, this logic ensures it would fail with an error telling the user they must update their fw